dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -830,6 +830,22 @@ public void rename(String src, String dst, RenamePOptions options)
     }
 
     try {
+      boolean overWrite = options.hasS3SyntaxOptions()
+          && options.getS3SyntaxOptions().hasOverwrite()
+          && options.getS3SyntaxOptions().getOverwrite();
+      if (dstUfs.exists(dst)) {
+        if (!overWrite) {
+          throw new RuntimeException(
+              new FileAlreadyExistsException("File already exists but no overwrite flag"));
+        } else {
+          mMetaManager.removeFromMetaStore(dst);
+          if (dstUfs.getStatus(dst).isFile()) {
```

Comment:
too many ufs call will lead to many overhead. `ufs.exists` `ufs.getStatus` minimize real ufs calls as much as possible